hkmc2/shared/src/test/mlscript/handlers/RecursiveHandlers.mls

```diff
@@ -0,0 +1,123 @@
+:js
+:handler
+
+
+abstract class Effect[A] with
+  fun perform(arg: A): A
+
+
+handle h1 = Effect with
+  fun perform(arg)(k) =
+    print("performing " + arg)
+    [k("ok")]
+h1.perform("hi")
+h1.perform("hello")
+//│ > performing hi
+//│ > performing hello
+//│ = [ [ 'ok' ] ]
+
+h1
+//│ = Effect$Effect$131 {}
+
+h1.perform("oops")
+//│ > <ref *1> __EffectSig {
+//│ >   next: null,
+//│ >   nextHandler: null,
+//│ >   tail: [Circular *1],
+//│ >   tailHandler: [Circular *1],
+//│ >   resumed: false,
+//│ >   handler: Effect$Effect$131 {},
+//│ >   handlerFun: [Function (anonymous)]
+//│ = }
+
+
+:fixme
+handle h2 = Effect with
+  fun perform(arg)(k) =
+    print("performing " + arg)
+    k of if arg > 0
+      then h2.perform(arg - 1) + " " + arg
+      else "0"
+[
+  print("–––");
+  h2.perform(2)
+  print("–––");
+  h2.perform(3)
+]
+//│ ╔══[ERROR] Name not found: h2
+//│ ║  l.39: 	      then h2.perform(arg - 1) + " " + arg
+//│ ╙──      	           ^^
+//│ > –––
+//│ > performing 2
+//│ > –––
+//│ > performing 3
+//│ = [ undefined, undefined ]
+
+
+let res =
+  handle h = Effect with
+    fun perform(arg)(k) =
+      print("A " + arg)
+      handle g = Effect with
+        fun perform(arg1)(k1) =
+          print("B " + arg + " " + arg1)
+          ["‹", k1(arg), arg1, "›"]
+      [k(g)]
+  let g = h.perform("hi")
+  g.perform("bye")
+  g.perform("friend")
+  set g = h.perform("hola")
+  g.perform("adios")
+  g.perform("amigos")
+in res.toString()
+//│ > A hi
+//│ > B hi bye
+//│ > B hi friend
+//│ > A hola
+//│ > B hola adios
+//│ > B hola amigos
+//│ = '‹,‹,‹,‹,hola,friend,›,bye,›,amigos,›,adios,›'
+
+let res =
+  print("A " + "hi")
+  handle g = Effect with
+    fun perform(arg1)(k1) =
+      print("B " + "hi" + " " + arg1)
+      ["‹", k1("hi"), arg1, "›"]
+  g.perform("bye")
+  g.perform("friend")
+  print("A " + "hola")
+  handle g2 = Effect with
+    fun perform(arg1)(k1) =
+      print("B " + "hola" + " " + arg1)
+      ["‹", k1("hola"), arg1, "›"]
+  g2.perform("adios")
+  g2.perform("amigos")
+in res.toString()
+//│ > A hi
+//│ > B hi bye
+//│ > B hi friend
+//│ > A hola
+//│ > B hola adios
+//│ > B hola amigos
+//│ = '‹,‹,‹,‹,hola,amigos,›,adios,›,friend,›,bye,›'
+
+
+:handler 
+let str = ""
+if true do
+  handle h1 = Effect with
+    fun perform(arg)(k) =
+      set str = str + "A"
```

Comment:
```suggestion
      set str += "A"
```

Reply:
Oh I didn't realize we had this operator, nice

Reply:
It works for all symbolic operators ;^)

BTW there are many places to update, not just the one I marked here.